### PR TITLE
Feature/test load groundtruth raster

### PR DESCRIPTION
**Pull request by hasruleng**
finished test-load_groundtruth_raster and moved test_data under tests

always adjust both config.yml, 
1. one in testthat to run unit test, 
2. one in root to run package/source code

### CODE REVIEW

File: R/ff_run.R [hunk 1]
```diff
@@ -173,14 +173,14 @@ ff_run <- function(shape = NULL, country = NULL, prediction_dates = NULL,
...
-    traindata <- do.call(ff_prep, ff_prep_params_combined)
+    traindata <- do.call(ff_prep_refactored, ff_prep_params_combined)
     if (hasvalue(validation_dates)) {
       if (verbose) {
         ff_cat("adding validation matrix for dates", paste(validation_dates, collapse = ", "), "\n", color = "green")
       }
       ff_prep_params_combined["dates"] <- validation_dates
       ff_prep_params_combined["sample_size"] <- 1 / 3 * sample_size
-      valdata <- do.call(ff_prep, ff_prep_params_combined)
+      valdata <- do.call(ff_prep_refactored, ff_prep_params_combined)
```
Comment (aminbksi) [inline R/ff_run.R:183]: change all ff_preps I dont comment for all of them

File: R/filter_by_feature.R [hunk 2]
```diff
@@ -28,12 +28,19 @@ filter_by_feature <- function(fltr_features, fltr_condition, matrix, verbose = T
       operator <- gsub("[[:alnum:]]", "", fltr_condition[i])
       value <- gsub("[^0-9]", "", fltr_condition[i])
       filtercolumn <- which(colnames(matrix) == fltr_features[i])
+      if (verbose) {
+        cat(paste("filtering features: before if (length(filtercolumn) == 0)\n"))
+      }
```
Comment (aminbksi) [inline R/filter_by_feature.R:33]: did you add this?

File: R/filter_by_feature.R [hunk 3]
```diff
@@ -28,12 +28,19 @@ filter_by_feature <- function(fltr_features, fltr_condition, matrix, verbose = T
...
       if (length(filtercolumn) == 0) {
-        ff_cat("The feature", fltr_features[i], "was not found, skipping filtering for this feature", color = "yellow")
+        ff_cat("The feature", fltr_features[i], "was not found, skipping filtering for this feature\n", color = "yellow")
       } else {
         if (verbose) {
           cat(paste("filtering feature", fltr_features[i], "on", fltr_condition[i], "\n"))
         }
+        
+        if (verbose) {
```
Comment (aminbksi) [inline R/filter_by_feature.R:41]: you added this?

File: R/filter_by_feature.R [hunk 4]
```diff
@@ -53,12 +60,18 @@ filter_by_feature <- function(fltr_features, fltr_condition, matrix, verbose = T
           sf_indices <- which(matrix[, filtercolumn] <= value)
         }
       }
+      if (verbose) {
```
Comment (aminbksi) [inline R/filter_by_feature.R:63]: ?

File: R/filter_by_feature.R [hunk 5]
```diff
@@ -53,12 +60,18 @@ filter_by_feature <- function(fltr_features, fltr_condition, matrix, verbose = T
...
+        print(paste("length sfa_indices:", length(sfa_indices),"\n", sfa_indices))
+      }
       if (length(sfa_indices) == 0) {
         sfa_indices <- c(sfa_indices, sf_indices)
       } else {
         sfa_indices <- intersect(sfa_indices, sf_indices)
       }
     }
+    if (verbose) {
```
Comment (aminbksi) [inline R/filter_by_feature.R:72]: ?

File: config.yml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 URL_MARKDOWN: "https://raw.githubusercontent.com/jduijvenbode-WWF/ForestForesight/main/README.md"
-AWS_BUCKET_NAME: "forestforesight-public"
+AWS_BUCKET_NAME: "forestforesight-publick"
```
Comment (aminbksi) [inline config.yml:2]: ??

**Review by aminbksi — CHANGES_REQUESTED** [review]

File: R/ff_run.R [hunk 7]
```diff
@@ -173,14 +173,14 @@ ff_run <- function(shape = NULL, country = NULL, prediction_dates = NULL,
     }
     ff_prep_params_combined <- merge_lists(ff_prep_params_original, ff_prep_params)
 
-    traindata <- do.call(ff_prep, ff_prep_params_combined)
+    traindata <- do.call(ff_prep_refactored, ff_prep_params_combined)
```
[the remark this answers is not in the document]
Comment (hasruleng) [reply R/ff_run.R:176]: OK

File: R/filter_by_feature.R (hunk 2 above)
Comment (hasruleng) [reply R/filter_by_feature.R:33]: yes, this is where test_data will fail.
I'll remove this.

**Review by aminbksi — APPROVED** [review]

**Pull request merged**